lib/publisher.rb

```diff
@@ -0,0 +1,44 @@
+class Publisher
+  def self.publish(presenter)
+    new(presenter).publish
+  end
+
+  attr_reader :presenter
+
+  def initialize(presenter)
+    @presenter = presenter
+  end
+
+  def publish
+    response = send_item_to_draft_stack
+    response = publish_links if presenter.links && response_successful?(response)
+    response = publish_item if response_successful?(response)
+    response
```

Comment:
I added the response as a return value because the code is checked in [redirector_for_gone_contact.rb](https://github.com/alphagov/contacts-admin/blob/upgrade-to-publishing-api-v2/lib/redirector_for_gone_contact.rb#L87-L89). That is the only place it is used though...